packages/core/src/App/Components/Routes/binary-routes.jsx

```diff
@@ -1,17 +1,25 @@
 import React from 'react';
-import { Switch } from 'react-router-dom';
+import { Switch, Prompt } from 'react-router-dom';
 import { Loading } from '@deriv/components';
+import { connect } from 'Stores/connect';
 import getRoutesConfig from 'App/Constants/routes-config';
 import RouteWithSubRoutes from './route-with-sub-routes.jsx';
 
 const BinaryRoutes = props => (
     <React.Suspense fallback={<Loading />}>
         <Switch>
             {getRoutesConfig().map((route, idx) => (
-                <RouteWithSubRoutes key={idx} {...route} {...props} />
+                <RouteWithSubRoutes
+                    key={idx}
+                    {...route}
+                    {...props}
+                    {...(route.title === 'Bot' && { prompt_before_leave: props.is_bot_running })}
```

Comment:
Please do not use `route.title` like this. You can use the `route.path` property if you need to check for path.